user_guide_src/source/libraries/validation/030.php

```diff
@@ -1,7 +1,7 @@
-<div class="alert alert-danger" role="alert">
-    <ul>
-    <?php foreach ($errors as $error): ?>
-        <li><?= esc($error) ?></li>
+<?php if (! empty($errors)): ?>
+    <div class="alert alert-danger">
+    <?php foreach ($errors as $field => $error): ?>
+        <p><?= $error ?></p>
```

Comment:
Why do you remove `esc()`?
Before: `esc($error)`
After: `$error`

Reply:
have not paid attention to it. Should of course contain `esc()`